pkg/entity/entity.go

```diff
@@ -18,17 +19,26 @@ const (
 )
 
 type Entity struct {
-	Loc  [2]int // s[Loc[0]:Loc[1]]
-	Type EntityType
+	Loc  [2]int     `json:"-"` // s[Loc[0]:Loc[1]]
+	Type EntityType `json:"type"`
 
-	Text string
-	Info string
+	Text     string      `json:"ip"`
+	InfoText string      `json:"text"`
+	Info     interface{} `json:"info"`
 }
 
 func (e Entity) ParseInfo() error {
 	return nil
 }
 
+func (e Entity) Json() string {
+	jsonResult, err := json.Marshal(e)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
```

Comment:
It may not be necessary to output the error in json format, but simply return exit(1) and output the error message to stderr, because once the program makes an error, subsequent processing will stop.